lib/Conversion/LTLToCore/LTLToCore.cpp

```diff
@@ -0,0 +1,237 @@
+#include "circt/Dialect/SV/SVDialect.h"
+#include "circt/Dialect/SV/SVOps.h"
+#include "circt/Dialect/Seq/SeqOps.h"
+#include "circt/Dialect/Verif/VerifOps.h"
+#include "circt/Support/BackedgeBuilder.h"
+#include "circt/Support/Namespace.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace hw;
+
+//===----------------------------------------------------------------------===//
+// Conversion patterns
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Lower a ltl::DisableOp operation to Core operations
+struct DisableOpConversion : OpConversionPattern<ltl::DisableOp> {
+  using OpConversionPattern<ltl::DisableOp>::OpConversionPattern;
+
+  // DisableOp translates to an implication in the form of
+  // ~condition -> input
+  LogicalResult
+  matchAndRewrite(ltl::DisableOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    // Replace the ltl::DisableOp with an OR op as it represents a disabling
+    // implication: (implies (not condition) input) is equivalent to
+    // (or (not (not condition)) input) which becomes (or condition input)
+    rewriter.replaceOpWithNewOp<comb::OrOp>(op, adaptor.getCondition(),
+                                            adaptor.getInput());
+
+    return success();
+  }
+};
+
+struct HasBeenResetOpConversion : OpConversionPattern<verif::HasBeenResetOp> {
+  using OpConversionPattern<verif::HasBeenResetOp>::OpConversionPattern;
+
+  // HasBeenReset generates a 1 bit register that is set to one if the reset
+  // value is found to be active at any point
+  LogicalResult
+  matchAndRewrite(verif::HasBeenResetOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    // Generate the constant used to set the register value
+    Value constZero = rewriter.create<hw::ConstantOp>(
+        op.getLoc(), mlir::IntegerType::get(op.getContext(), 1), 0);
+
+    // Create a backedge for the register to be used in the mux
+    circt::BackedgeBuilder bb(rewriter, op.getLoc());
+    circt::Backedge reg = bb.get(rewriter.getI1Type());
+
+    // Generate an or between the reset and the register's value to store
+    // whether or not the reset has been active at least once
+    Value orReset =
+        rewriter.create<comb::OrOp>(op.getLoc(), adaptor.getReset(), reg);
+
+    // This register should not be reset, so we give it dummy reset and resetval
+    // operands to fit the build signature
+    Value reset, resetval;
+
+    // Finally generate the register to set the backedge
+    reg.setValue(rewriter.replaceOpWithNewOp<seq::CompRegOp>(
+        op, orReset, adaptor.getClock().getDefiningOp()->getOperand(0), reset,
+        resetval, llvm::StringRef("hbr"), constZero));
```

Comment:
ok, I wasn't sure how to generate readable name uniquely. Ideally I would like it to be prefixed with `hbr` or something related so that it can easily be distinguished in the output SV if people decide to go that route. Is there a good way to do this?